src/DataWrangling/EN4/EN4.jl

```diff
@@ -0,0 +1,257 @@
+export EN4Monthly
+export EN4FieldTimeSeries, EN4Restoring, LinearlyTaperedPolarMask
+
+using ClimaOcean
+using ClimaOcean.DataWrangling
+using ClimaOcean.DataWrangling: inpaint_mask!, NearestNeighborInpainting, download_progress, compute_native_date_range
+using ClimaOcean.InitialConditions: three_dimensional_regrid!, interpolate!
+
+using Oceananigans
+using Oceananigans: location
+using Oceananigans.Architectures: architecture, child_architecture
+using Oceananigans.BoundaryConditions
+using Oceananigans.DistributedComputations
+using Oceananigans.DistributedComputations: DistributedField, all_reduce, barrier!
+using Oceananigans.Utils
+
+using KernelAbstractions: @kernel, @index
+using NCDatasets
+using JLD2
+using Downloads: download
+using Dates
+using Adapt
+using Scratch
+
+download_EN4_cache::String = ""
+function __init__()
+    global download_EN4_cache = @get_scratch!("EN4")
+end
+
+include("EN4_metadata.jl")
+include("EN4_mask.jl")
+
+# Vertical coordinate
+const EN4_z = [
+    -5500.0,
+    -5200.5986,
+    -4901.459,
+    -4602.695,
+    -4304.469,
+    -4007.0154,
+    -3710.6636,
+    -3415.8828,
+    -3123.3313,
+    -2833.926,
+    -2548.9243,
+    -2270.0168,
+    -1999.4135,
+    -1739.8875,
+    -1494.7288,
+    -1267.542,
+    -1061.846,
+    -880.5102,
+    -725.1709,
+    -595.8501,
+    -490.9494,
+    -407.6244,
+    -342.3651,
+    -291.5638,
+    -251.9186,
+    -220.6418,
+    -195.5097,
+    -174.8189,
+    -157.3019,
+    -142.0345,
+    -128.3522,
+    -115.7825,
+    -103.9913,
+    -92.7437,
+    -81.8753,
+    -71.2711,
+    -60.8509,
+    -50.5586,
+    -40.3553,
+    -30.214,
+    -20.1158,
+    -10.0475,
+    -0.0,
+]
+
+empty_EN4_field(variable_name::Symbol; kw...) = empty_EN4_field(Metadatum(variable_name, dataset=EN4Monthly()); kw...)
+
+function empty_EN4_field(metadata::EN4Metadata;
+                          architecture = CPU(), 
+                          horizontal_halo = (7, 7))
+
+    Nx, Ny, Nz, _ = size(metadata)
+    loc = location(metadata)
+    longitude = (0, 360)
+    latitude = (-90, 90)
+    TX, TY = (Periodic, Bounded)
+
+    if variable_is_three_dimensional(metadata)
+        TZ = Bounded
+        LZ = Center
+        z = EN4_z
+        halo = (horizontal_halo..., 3)
+        sz = (Nx, Ny, Nz)
+    else # the variable is two-dimensional
+        TZ = Flat
+        LZ = Nothing
+        z = nothing
+        halo = horizontal_halo
+        sz = (Nx, Ny)
+    end
+
+    grid = LatitudeLongitudeGrid(architecture, Float32; halo, longitude, latitude, z,
+                                 size = sz,
+                                 topology = (TX, TY, TZ))
+
+    return Field{loc...}(grid)
+end
+
+# Only temperature and salinity need a thorough inpainting because of stability,
+# other variables can do with only a couple of passes. Sea ice variables 
+# cannot be inpainted because zeros in the data are physical, not missing values.
+function default_inpainting(metadata::EN4Metadata)
```

Comment:
I think if this is the same for ECCO and EN4, then perhaps it can be the fallback method

Reply:
Yes these are the same.

Reply:
Yes, no need for a new method then.
I can help smooth things out.

You understand what fall back is?

@glwagner suggests something:

```Julia
function default_inpainting(metadata)
    do_things
    return some_things
end
```

and if particular datasets want something different they can implement


```julia
function default_inpainting(metadata::EN4Metadata)
    do_special_EN4_things
    return some_things
end
```